File: amazia-core/src/main/java/com/example/product/controller/UpdateProductController.java
package com.example.product.controller;

import com.example.product.entity.Product;
import com.example.product.service.UpdateProductService;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class UpdateProductController {

    private final UpdateProductService updateProductService;

    public UpdateProductController(UpdateProductService updateProductService) {
        this.updateProductService = updateProductService;
    }

    @PutMapping("/products/{id}")
    public ResponseEntity<Product> update(@PathVariable Long id, @Valid @RequestBody Product request) {
        return ResponseEntity.ok(updateProductService.update(id, request));
    }
}

File: amazia-console/app/Product/Controller/UpdateProductController.php
<?php

namespace App\Product\Controller;

use App\Http\Controllers\Controller;
use App\Product\Service\UpdateProductService;
use Illuminate\Http\Request;

class UpdateProductController extends Controller
{
    private UpdateProductService $service;

    public function __construct(UpdateProductService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request, int $id)
    {
        $response = $this->service->update($id, $request->all());
        return response()->json($response->json(), $response->status());
    }
}
